stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,28 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
+    let repeatedValueGuts = repeatedValue._guts
+    let buffer: UnsafeMutableBufferPointer<UInt8> =
+      .allocate(capacity: repeatedValueGuts.count &* count)
+    defer { buffer.deallocate() }
+    var offset = 0
     for _ in 0..<count {
-      result += repeatedValue
+      let bufferRebased = UnsafeMutableBufferPointer(rebasing: buffer[offset...])
+      guard let copied = repeatedValueGuts.copyUTF8(into: bufferRebased) else {
+        _internalInvariantFailure("Buffer's capacity \(buffer.count)" +
+          "is not enough to accommodate '\(repeatedValue)' \(count) times")
+      }
+      offset += copied
     }
-    self = result
+    _internalInvariant(offset == buffer.count)
+    self = repeatedValueGuts.isASCII ?
+      String._uncheckedFromASCII(UnsafeBufferPointer(buffer)) :
+      String._uncheckedFromUTF8(UnsafeBufferPointer(buffer), isASCII: false)
```

Comment:
Do we need this ternary logic here, or does `_uncheckedFromUTF8` do this anyway using the `isASCII` argument?

Reply:
fixed